src/services/accounts/AccountsBalanceInfoService.ts

```diff
@@ -161,14 +186,65 @@ export class AccountsBalanceInfoService extends AbstractService {
 				at,
 				nonce,
 				tokenSymbol: token,
-				free,
-				reserved,
-				miscFrozen,
-				feeFrozen,
-				locks,
+				free: withDenomination ? this.applyDenomination(free, decimal) : free,
+				reserved: withDenomination
+					? this.applyDenomination(reserved, decimal)
+					: reserved,
+				miscFrozen: withDenomination
+					? this.applyDenomination(miscFrozen, decimal)
+					: miscFrozen,
+				feeFrozen: withDenomination
+					? this.applyDenomination(feeFrozen, decimal)
+					: feeFrozen,
+				locks: withDenomination ? this.denominateLocks(locks, decimal) : locks,
 			};
 		} else {
 			throw new BadRequest('Account not found');
 		}
 	}
+
+	/**
+	 * Apply a denomination to a balance depending on the chains decimal value.
+	 *
+	 * @param balance free balance available encoded as Balance
+	 * @param dec The chains given decimal value
+	 */
+	private applyDenomination(balance: Balance, dec: number): string {
+		const strBalance = balance.toString();
+
+		// We dont want to denominate a zero balance
+		if (strBalance === '0') {
+			return strBalance;
+		}
+		// If the denominated value will be less then zero, pad it correctly
+		if (strBalance.length <= dec) {
```

Comment:
I'm struggling to parse exactly how thse logic for this is working, but I can see that it's being tested so it's not a big deal, just curious really!

Reply:
Well the idea was how can we properly denominate the decimal places without ever having the string (number represented as a string) be converted to a number. This is because the atomic value given can be too large for javascript to interpret. So in the logic above I wanted to make sure we also never have to BN, or convert to a bigInt. We can easily fix the value to a decimal with some simple math, but because the values might be to big thats not possible.

So basically what the logic does takes care of three cases.

1. If the value is zero just return zero

2. If the decimal value is larger than the length of the atomic value (it's a string), the value denominated is going to be less than zero. Which means it needs the correct amount of padding (zeros) after the decimal. 

3. The value is greater than zero, and therefore handle the decimal places by subtracting the length of the value, by the decimal places, and using the value to determine where we put the decimal.

Reply:
> If the decimal value is larger than the length of the atomic value (it's a string), the value denominated is going to be less than zero. Which means it needs the correct amount of padding (zeros) after the decimal.

I don't follow that but maybe I'm just slow such you might have a number like `123.9999` (decimal is bigger than the number) why do you need to pad it?

Reply:
I also had non-integers in mind, but makes sense to me after both of your comments; thanks!

Reply:
> ah, right the decimals is just some number such as 10 that the number should be multiplied with not an actual decimal point. Sorry I got confused by dec.

Yup exactly!

Important to note as well, that the balance passed in will always be represented as an atomic balance via the `Balance` type.